### ISSUE TITLE
"A task was canceled" on connecting to Google - A Lot!

### ISSUE THREAD

**Issue by plevintampabay** [phw198/OutlookGoogleCalendarSync#1300]
> Checklist:-
> * [ ] [Upgrade](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Upgrading) to the latest release (alpha if available) ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg)
> * [ ] Reproduce the issue
> * [ ] Complete the information below
> * [ ] Provide your `OGcalsync.log` file (see [how-to](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#locating-your-ogcalsynclog-file))

**OGCS Version**: v2.9.0.0
**Edition**: Installed 
**Sync Direction**:  O<->G

**Bug Description**  
I Just installed your app yesterday so I hope I have the latest version.
I am very impressed, but you have one big problem, I think you may already know about.
You often fail to connect to Google.
At line 237 in the attached log file is the last time it failed to connect.
It also crashed this morning - I sent in the log file anonymously.

**Steps to Reproduce the Issue**  
Just let it automatically update, currently set for once per hour

**Any other information**  
none
:paperclip: Attach your `OGcalsync.log` file.
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/7245841/OGcalsync.log)

**Comment by phw198** [comment]
I can't really make out what going on with your logfile - there seem to be missing chunks as it's logging and finishes a sync, then there's no sign of a shutdown, but 3 minutes later you've started OGCS again and somehow logging is configured to be OFF! Did you terminate the program with Task Manager or something and then edit the settings.xml file manually??

```
2021-09-28 11:19:33,226 DEBUG  1 OutlookGoogleCalendarSync.XMLManager [135] -  Setting 'CompletedSyncs' updated to '10'
2021-09-28 11:19:33,227 INFO  1 OutlookGoogleCalendarSync.Console [282] -  Sync finished!
2021-09-28 11:19:33,244 DEBUG  1 OutlookGoogleCalendarSync.XMLManager [135] -  Setting 'LastSyncDate' updated to '9/28/2021 11:17:46 AM'
2021-09-28 11:19:33,245 INFO  1 OutlookGoogleCalendarSync.Sync.SyncTimer [18] -  Next sync scheduled for Tuesday, September 28, 2021 @ 12:17:46 PM
2021-09-28 11:22:20,821 INFO  1 OutlookGoogleCalendarSync.GoogleOgcs.ErrorReporting [178] -  Initialising error reporting to OFF
2021-09-28 11:22:20,837 INFO  1 OutlookGoogleCalendarSync.Program [242] -  Program started: v2.9.0.0
2021-09-28 11:22:20,837 INFO  1 OutlookGoogleCalendarSync.Program [243] -  Started interactively.
...
2021-09-28 11:22:20,847 DEBUG  1 OutlookGoogleCalendarSync.XMLManager [58] -  Retrieved setting 'LoggingLevel' with value 'OFF'
2021-09-28 11:22:20,848 INFO  1 OutlookGoogleCalendarSync.Settings [603] -  Logging level configured to 'OFF'
```
Because logging is off, I can't see if you're using a proxy or something, but
* at 1am it determines there is no internet
* I then suspect your computer goes to sleep, woken up around 7am
* 7:37 sync works OK, next sync scheduled for 8:37
* Looks like computer goes to sleep again until 10:23
* 10:38 sync fails to talk to Google, next sync scheduled 11:17
* 11:17 sync works, but there's a 2 minute delay with the first call to Google

Basically, Google is cancelling your request on you because I think you have [network problems](https://github.com/googleapis/google-api-dotnet-client/issues/1394). 
```

2021-09-28 10:38:03,972 INFO  15 OutlookGoogleCalendarSync.Console [282] -  Scanning Google calendar...
2021-09-28 10:39:43,995 WARN  15 OutlookGoogleCalendarSync.OGCSexception [1412] -  Not able to retrieve retrieve Google calendar's global timezone
2021-09-28 10:39:43,996 ERROR  15 OutlookGoogleCalendarSync.OGCSexception [28] -  System.Threading.Tasks.TaskCanceledException: A task was canceled.
```
Fix the network problem and OGCS should be able to communicate with Google better...

**Comment by plevintampabay** [comment]
This issue is working better now
